Speed up seeds

deivid-rodriguez opened the pull request:
#### :tophat: What? Why?

I added a few small changes to speed up seeds a bit. I don't think we need as many objects as we used to have. What do you think?

* Before:

```console
$ time bundle exec rails db:seed

real	1m22.813s
user	1m16.380s
sys	0m2.984s
```

* After:

```console
$ time bundle exec rails db:seed

real	0m28.093s
user	0m24.944s
sys	0m1.948s
```

What do you 
#### :pushpin: Related Issues
- Related to #1921.

#### :clipboard: Subtasks
_None_.

### :camera: Screenshots (optional)
_None_.

#### :ghost: GIF
![referendum_one_week](https://user-images.githubusercontent.com/2887858/30807835-d1c91870-a1fb-11e7-9cb6-a57da92f85c9.gif)

mrcasals commented:
Looks good to me, but I'm not sure on the budgets part, changes could make us have a single budget with no attachments, and we might need to have at least one to ensure design looks good. Can you make budgets have at least one attachment?

oriolgual commented:
I'm OK with this as long as we make sure we always create a minimum number of examples.

deivid-rodriguez commented:
@mrcasals I restored attachments for all projects. I also added other improvements although they didn't speed it up as much as I expected.

Now it's

```console
$ time bundle exec rails db:seed

real	0m24.253s
user	0m21.632s
sys	0m1.820s
```

 Of course, there's randomness here. I wonder if there's a benefit in the randomness by the way? If we removed it we could have "idempotent" seeds which is always nice (meaning that running them twice won't crash and will always leave the DB in a consistent state).

deivid-rodriguez commented:
> I'm OK with this as long as we make sure we always create a minimum number of examples.

So maybe I should change `rand(3)` with `rand(3) + 1`? Although I'd honestly just go with `2.times` because of my previous comment.

mrcasals commented:
Yes, I think `rand` can return 0:

> When `max` is an Integer, rand returns a random integer greater than or equal to zero and less than `max`.
>
>       https://ruby-doc.org/core-2.2.0/Random.html#method-i-rand

So `rand(3)` can return one of `[0, 1, 2]`. So `rand(3).times` can return `0.times`, which does not run at all. Can we make sure the loops run at least once?

mrcasals commented:
@deivid-rodriguez +1 to move it to `2.times`, or even maybe just creating one of the elements (instead of two of them)

deivid-rodriguez commented:
I added a few more changes, now it's

```console
$ time bundle exec rails db:seed

real	0m21.550s
user	0m19.160s
sys	0m1.928s
```

Not sure of I went too far or if I should go further... :stuck_out_tongue_closed_eyes:

josepjaume commented:
+1 on everything that's being said here 💃

deivid-rodriguez commented:
So is this ready or should I "go minimal" and create things just once instead of `2.times`?

mrcasals commented:
@deivid-rodriguez I think it's good as it is, having 2 elements lets us check filters and grid layouts so it's good for me.

deivid-rodriguez commented:
Good point! :+1:

josepjaume commented on decidim-proposals/lib/decidim/proposals/feature.rb:
```diff
@@ -101,14 +101,14 @@
       user_group = [true, false].sample ? author.user_groups.verified.sample : nil
-      state, answer = if n > 15
+      state, answer = if n > 4
                         ["accepted", Decidim::Faker::Localized.sentence(10)]
```
Looks like `n` is never be more than `4` if we do stuff 5 times :D. Thanks Codecov!

deivid-rodriguez commented:
It's ready now!
